crates/hir_def/src/generics.rs

```diff
@@ -279,8 +294,31 @@ impl GenericParams {
             } else {
                 continue;
             };
+
+            let lifetimes: Option<Box<_>> = pred.generic_param_list().map(|param_list| {
+                // Higher-Ranked Trait Bounds
+                param_list
+                    .lifetime_params()
+                    .map(|lifetime_param| {
+                        let name = lifetime_param
+                            .lifetime()
+                            .map_or_else(Name::missing, |lt| Name::new_lifetime(&lt));
+                        let param = LifetimeParamData { name: name.clone() };
+                        // FIXME: allocating the lifetime here means it looks like this lifetime belongs to the entire item
+                        // but it only belongs to the bounded type-param
```

Comment:
Yeah, I'm not sure giving these lifetimes a `LifetimeParamId` at all is the correct approach, since they're not parameters of the item. (There's another place where new lifetime names can be defined, namely `for<'x> fn()` function pointer types, and I think these two should work the same.)

Reply:
Also, while shadowing an existing lifetime is disallowed, you can easily have the same lifetime twice on an item this way:
```rust
fn foo<T, U>() where for<'b> U: 'b, for<'b> T: 'b {}
```

Reply:
Well the main reason for giving them `LifetimeParamId`s is that for reference search we need to be able to address them as `code_model::LifetimeParam`s, which is backed by a LifetimeParamId. And technically they are LifetimeParams right? Just with a smaller scope. Though I can't really think of a way to solve this problem, so I guess it's better to just not support reference search for these for the time being.

Reply:
Yeah, I don't really know how to model this yet either, and I think supporting reference search for these is probably not super important right now.

Reply:
Yea I agree, will remove the FIXME parts for now then. 👍